<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PyriteEngineClass</class>
 <widget class="QMainWindow" name="PyriteEngineClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Pyrite Editor</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>Resources/Pyrite.png</normaloff>Resources/Pyrite.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QMdiArea" name="mdiArea">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1920</width>
      <height>1080</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Scene</string>
    </property>
    <widget class="QMenu" name="menuAdd_Trigger">
     <property name="title">
      <string>Add Trigger</string>
     </property>
     <addaction name="actionSquare"/>
    </widget>
    <addaction name="actionAdd_Object"/>
    <addaction name="menuAdd_Trigger"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionHelp"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionRun_Program"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuWindow"/>
   <addaction name="menuAbout"/>
   <addaction name="menuDebug"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="AssetToScene" name="RenderZone">
   <property name="minimumSize">
    <size>
     <width>1366</width>
     <height>38</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>1366</width>
     <height>768</height>
    </size>
   </property>
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <property name="accessibleName">
    <string>RenderZone</string>
   </property>
   <property name="windowTitle">
    <string>Scene</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents"/>
  </widget>
  <widget class="QDockWidget" name="Assets">
   <property name="minimumSize">
    <size>
     <width>1920</width>
     <height>259</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>1920</width>
     <height>259</height>
    </size>
   </property>
   <property name="accessibleName">
    <string>Assets</string>
   </property>
   <property name="windowTitle">
    <string>Assets</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <widget class="DragAndDrop" name="listWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>1920</width>
       <height>251</height>
      </rect>
     </property>
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
     <property name="dragDropMode">
      <enum>QAbstractItemView::DragDrop</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="movement">
      <enum>QListView::Free</enum>
     </property>
     <property name="flow">
      <enum>QListView::LeftToRight</enum>
     </property>
     <property name="viewMode">
      <enum>QListView::IconMode</enum>
     </property>
     <property name="uniformItemSizes">
      <bool>true</bool>
     </property>
     <property name="selectionRectVisible">
      <bool>true</bool>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="Components">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>548</width>
     <height>761</height>
    </size>
   </property>
   <property name="toolTipDuration">
    <number>-1</number>
   </property>
   <property name="accessibleName">
    <string>Components</string>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="windowTitle">
    <string>Components</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QWidget" name="ComponentWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="ObjectProps">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="accessibleName">
           <string>ObjectProps</string>
          </property>
          <property name="title">
           <string>Object Properties</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="2" column="6">
            <widget class="QDoubleSpinBox" name="RotZ">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="RotX">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Scale</string>
             </property>
            </widget>
           </item>
           <item row="2" column="4">
            <widget class="QDoubleSpinBox" name="RotY">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="5" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Z:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Position</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Size:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="5" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Z:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Y:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="4">
            <widget class="QDoubleSpinBox" name="PosY">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="1" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>X:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Y:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QDoubleSpinBox" name="PosX">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QDoubleSpinBox" name="Size">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Rotation</string>
             </property>
            </widget>
           </item>
           <item row="1" column="6">
            <widget class="QDoubleSpinBox" name="PosZ">
             <property name="minimum">
              <double>-1000000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1" alignment="Qt::AlignRight">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>X:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Object Name:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLineEdit" name="ObjectName">
             <property name="placeholderText">
              <string>ObjectName</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="AddAction" name="TransformBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Transform</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QPushButton" name="AddMovement">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Add Movement</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="CollisionBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="title">
           <string>Collision</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Collision Type: </string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="CollisionType"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="TriggerBox">
          <property name="title">
           <string>Trigger</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Connected Object:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ConnectedObject"/>
           </item>
           <item>
            <widget class="QPushButton" name="AddTriggerAction">
             <property name="text">
              <string>Add Action</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="ComponentSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="contextMenuPolicy">
           <enum>Qt::DefaultContextMenu</enum>
          </property>
          <property name="text">
           <string>Add Component</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionRun_Program">
   <property name="text">
    <string>Run Program</string>
   </property>
  </action>
  <action name="actionAdd_Object">
   <property name="text">
    <string>Add Object</string>
   </property>
  </action>
  <action name="actionSquare">
   <property name="text">
    <string>Square</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save Project As..</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load Project</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New Project</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <zorder>Assets</zorder>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DragAndDrop</class>
   <extends>QListWidget</extends>
   <header>draganddrop.h</header>
  </customwidget>
  <customwidget>
   <class>AssetToScene</class>
   <extends>QDockWidget</extends>
   <header>assettoscene.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AddAction</class>
   <extends>QGroupBox</extends>
   <header>addaction.h</header>
   <container>1</container>
   <slots>
    <slot>addToBox()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../PyriteEditor.qrc"/>
 </resources>
 <connections/>
</ui>
